File: src/main/java/org/jukeboxmc/inventory/CreativeItemCacheInventory.java
package org.jukeboxmc.inventory;

import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class CreativeItemCacheInventory extends Inventory{

    public CreativeItemCacheInventory( InventoryHolder holder ) {
        super( holder, 1 );
    }

    @Override
    public void sendContents( Player player ) {

    }

    @Override
    public void sendContents( int slot, Player player ) {

    }

    @Override
    public Player getInventoryHolder() {
        return (Player) this.holder;
    }

    @Override
    public @NotNull InventoryType getType() {
        return InventoryType.CREATIVE;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/inventory/CreativeItemCacheInventory.kt
package org.jukeboxmc.server.inventory

import org.jukeboxmc.api.inventory.InventoryHolder
import org.jukeboxmc.api.inventory.InventoryType
import org.jukeboxmc.server.player.JukeboxPlayer

class CreativeItemCacheInventory(
    inventoryHolder: InventoryHolder
) : JukeboxInventory(inventoryHolder, 1) {

    override fun getType(): InventoryType {
        return InventoryType.CREATIVE
    }

    override fun getInventoryHolder(): JukeboxPlayer {
        return super.getInventoryHolder() as JukeboxPlayer
    }

    override fun sendContents(player: JukeboxPlayer) {

    }

    override fun sendContents(slot: Int, player: JukeboxPlayer) {
    }

}